### PR TITLE
Manager Role Progress Plus Minor Refactors

### PR DESCRIPTION
**Pull request by jakesboy2**

### CODE REVIEW

File: src/Helpers/CreepHelper.ts [hunk 1]
```diff
@@ -118,6 +119,7 @@ export default class CreepHelper {
             return new RoomPosition(25, 25, creep.memory.job.targetID);
         } else if (creep.memory.job && creep.memory.job.targetType === "roomPosition") {
             // TODO Handle parsing a string into a roomPosition object here
+            // what the hell is line 125 lmfao. Also is this done? If not explain further
```
Comment (UhmBrock) [inline src/Helpers/CreepHelper.ts]: Not implemented

File: src/Helpers/MemoryHelper_Room.ts [hunk 2]
```diff
@@ -535,4 +538,45 @@ export default class MemoryHelper_Room {
         // * Optionally apply a filter or otherwise check the limits before assigning them
         Memory.rooms[room.name].creepLimit!.remoteLimits = newLimits;
     }
+
+    /**
+     * update the memory which contains the center of the bunker for the room
+     * @param room the room we are in
+     */
+    public static updateBunkerCenter(room: Room): void {
```
Comment (UhmBrock) [inline src/Helpers/MemoryHelper_Room.ts]: Not a big fan of how we do this right now. Likely to change

File: src/Helpers/RoleHelpers/ManagerBodyOptsHelper.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    ROOM_STATE_UPGRADER,
+    TIER_6,
+    TIER_7,
+    TIER_8,
+    ROLE_MANAGER,
+} from "utils/Constants";
+import { SpawnHelper } from "Helpers/SpawnHelper";
+import SpawnApi from "Api/Spawn.Api"
+
+export class MedicBodyOptsHelper implements ICreepBodyOptsHelper {
```
Comment (UhmBrock) [inline src/Helpers/RoleHelpers/ManagerBodyOptsHelper.ts]: Medic -> Manager

File: src/Jobs/ClaimPartJobs.ts [hunk 4]
```diff
@@ -188,9 +188,13 @@ export class ClaimPartJobs implements IJobTypeHelper {
     /**
      * Gets a list of SignJobs for the room (signing the controller)
      * @param room The room to get the jobs for
+     *
      */
     public static createSignJobs(room: Room): ClaimPartJob[] {
         // TODO Get a list of controllers to be signed
+        // Every controller we have vision of the room in the scouting structure,
+        // its not owned by another player,
```
Comment (UhmBrock) [inline src/Jobs/ClaimPartJobs.ts]: Oh solid callout. Good use of that data

File: src/Managers/Roles/ManagerCreepManager.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import MemoryApi from "../../Api/Memory.Api";
+import CreepApi from "Api/Creep.Api";
+import {
+    ROLE_MANAGER,
+} from "utils/constants";
+
+// Manager for the miner creep role
+export default class MinerCreepManager implements ICreepRoleManager {
```
Comment (UhmBrock) [inline src/Managers/Roles/ManagerCreepManager.ts]: Miner -> Manager

File: src/Helpers/MemoryHelper_Room.ts (hunk 2 above)
Comment (jakesboy2) [reply src/Helpers/MemoryHelper_Room.ts]: Extremely open to change on this one. Had a tough time figuring this out

**Pull request merged**